### PR TITLE
Fix soundpacks not working, auto install dependencies, defer logging in tests

### PR DESCRIPTION
**Pull request by pitaj**
- Fix linting fail
- Fix soundpacks not working
- Automatically install dependencies on first `./nodebb`
- Defer winston logging until after tests finish

### CODE REVIEW

File: nodebb [hunk 1]
```diff
@@ -403,16 +401,7 @@ var commands = {
...
-			var name = args._[1];
-			if (!name) {
-				process.stdout.write(commands.activate.usage + '\n');
-				process.exit();
-			}
-			if (name.startsWith('nodebb-theme')) {
-				fork(['--reset', '-t', name]);
-				return;
-			}
-			var arr = ['--activate=' + name].concat(process.argv.slice(4));
```
Comment (julianlam) [inline nodebb]: Why was this removed?

File: nodebb [hunk 2]
```diff
@@ -428,18 +417,6 @@ var commands = {
...
-				var arr = ['--upgrade'].concat(process.argv.slice(3));
-				var upgradeProc = fork(arr);
-
-				return upgradeProc.on('close', function (err) {
-					if (err) {
-						process.stdout.write('\nError'.red + ': ' + err.message + '\n');
-					}
-				});
-			}
-
```
Comment (julianlam) [inline nodebb]: Why was this removed?

File: nodebb [hunk 3]
```diff
@@ -2,36 +2,34 @@
...
-	require('colors');
-	cproc = require('child_process');
-	args = require('minimist')(process.argv.slice(2));
-	fs = require('fs');
-	path = require('path');
-	request = require('request');
-	semver = require('semver');
-	prompt = require('prompt');
-	async = require('async');
+	fs.readFileSync(path.join(__dirname, 'node_modules/async/package.json'));
```
Comment (julianlam) [inline nodebb]: This would only throw if async is not installed...

File: nodebb (hunk 1 above)
Comment (pitaj) [inline nodebb]: That wasn't showing up in the diff for me. Odd. I'll fix it.

File: nodebb (hunk 3 above)
Comment (pitaj) [inline nodebb]: Right, but I think that any call to `npm install` will install all packages, even if it's only called to install a certain plugin. So unless packages were deleted manually they're either all the or none are there, I think.

File: nodebb (hunk 2 above)
Comment (pitaj) [inline nodebb]: Accident

File: nodebb (hunk 3 above)
Comment (julianlam) [inline nodebb]: Unfortunately we can't trust npm with that. Just today `request` wasn't properly installed for me, and I needed to reinstall it.

Comment (pitaj) [inline nodebb]: Well would it be good to just check every dependency in `package.json`, then?

File: nodebb [hunk 4]
```diff
@@ -2,36 +2,52 @@
...
-		process.stdout.write('For more information, please see: https://docs.nodebb.org/en/latest/installing/os.html\n\n');
-		process.stdout.write('Could not start: ' + e.code + '\n');
+	process.stdout.write(
+		'\x1b[31mNodeBB could not be initialised because there was an error while loading dependencies.\n' +
+		'Please run "\x1b[33mnpm install --production\x1b[31m" and try again.\x1b[0m\n\n' +
+		'For more information, please see: https://docs.nodebb.org/en/latest/installing/os.html\n\n'
+	);
 
-		process.exit(1);
-	}
```
Comment (julianlam) [inline nodebb]: Needs an `else`, I think, otherwise a non-module-related error would just cause it to hang maybe?

Comment (pitaj) [inline nodebb]: You commented on the previous file. The new one will just `throw` on every error, there's no conditional now.

Comment (julianlam) [inline nodebb]: oh okay 👍

File: nodebb [hunk 5]
```diff
@@ -2,36 +2,52 @@
...
 	async = require('async');
 } catch (e) {
-	if (e.code === 'MODULE_NOT_FOUND') {
-		process.stdout.write('NodeBB could not be started because it\'s dependencies have not been installed.\n');
-		process.stdout.write('Please ensure that you have executed "npm install --production" prior to running NodeBB.\n\n');
-		process.stdout.write('For more information, please see: https://docs.nodebb.org/en/latest/installing/os.html\n\n');
-		process.stdout.write('Could not start: ' + e.code + '\n');
+	process.stdout.write(
+		'\x1b[31mNodeBB could not be initialised because there was an error while loading dependencies.\n' +
+		'Please run "\x1b[33mnpm install --production\x1b[31m" and try again.\x1b[0m\n\n' +
```
Comment (julianlam) [inline nodebb]: Oh, is this for colours? You can use `colors` module for that.

`'Please run "' + 'npm install --production'.yellow + ': and try again."`

Comment (pitaj) [inline nodebb]: Not if the colors module doesn't exist. Then it just writes out undefined.

Comment (julianlam) [inline nodebb]: Fair enough, thanks.
